paramiko/config.py

```diff
@@ -49,7 +50,7 @@ def __init__(self):
         """
         self._config = []
 
-    def parse(self, file_obj):
+    def parse(self, file_obj, parsed_files=[]):
```

Comment:
There's a potential gotcha here - if `parse()` is called on two separate occasions (non-recursively), the default value the second time will retain the modifications (appends) from the first time `parse()` was called.

Generally, whenever you want a default value for a param, and you mutate that param, you want it like so:

```python
def func(param=None):
    if param is None:
        param = []
```